Course/Course/Domain/Model/CourseDates.swift

```diff
@@ -0,0 +1,210 @@
+            return "Today"
+        } else {
+            return blockStatus.title
+        }
+    }
+    
+    var isInPast: Bool {
+        return date.isInPast
+    }
+    
+    var isToday: Bool {
+        if dateType.isEmpty {
+            return true
+        } else {
+            return date.isToday
+        }
+    }
+    
+    var isInFuture: Bool {
+        return date.isInFuture
+    }
+    
+    var isAssignment: Bool {
+        return BlockStatus.status(of: dateType) == .assignment
+    }
+    
+    var isVerifiedOnly: Bool {
+        return !learnerHasAccess
+    }
+    
+    var isComplete: Bool {
+        return complete ?? false
+    }
+    
+    var isLearnerAssignment: Bool {
+        return learnerHasAccess && isAssignment
+    }
+    
+    var isPastDue: Bool {
+        return !isComplete && (date < .today)
+    }
+    
+    var isUnreleased: Bool {
+        return link.isEmpty
+    }
+    
+    var canShowLink: Bool {
+        return !isUnreleased && isLearnerAssignment
+    }
+    
+    var blockStatus: BlockStatus {
+        if isComplete {
+            return .completed
+        }
+        
+        if !learnerHasAccess {
+            return .verifiedOnly
+        }
+        
+        if isAssignment {
+            if isInPast {
+                return isUnreleased ? .unreleased : .pastDue
+            } else if isToday || isInFuture {
+                return isUnreleased ? .unreleased : .dueNext
+            }
+        }
+        
+        return BlockStatus.status(of: dateType)
+    }
+}
+
+public struct DatesBannerInfo {
+    let missedDeadlines, contentTypeGatingEnabled, missedGatedContent: Bool
+    let verifiedUpgradeLink: String?
+}
+
+public enum BlockStatus {
+    case completed
+    case pastDue
+    case dueNext
+    case unreleased
+    case verifiedOnly
+    case assignment
+    case verifiedUpgradeDeadline
+    case courseExpiredDate
+    case verificationDeadlineDate
+    case certificateAvailbleDate
+    case courseStartDate
+    case courseEndDate
+    case event
+    
+    static func status(of type: String) -> BlockStatus {
+        switch type {
+        case "assignment-due-date": return .assignment
+        case "verified-upgrade-deadline": return .verifiedUpgradeDeadline
+        case "course-expired-date": return .courseExpiredDate
+        case "verification-deadline-date": return .verificationDeadlineDate
+        case "certificate-available-date": return .certificateAvailbleDate
+        case "course-start-date": return .courseStartDate
+        case "course-end-date": return .courseEndDate
+        default: return .event
+        }
+    }
+    
+    var title: String {
+        switch self {
+        case .completed:
+            return "Completed"
+        case .pastDue:
+            return "Past due"
+        case .dueNext:
+            return "Due next"
+        case .unreleased:
+            return "Unreleased"
+        case .verifiedOnly:
+            return "Verified Only"
```

Comment:
Same to these.
It would be great to move them to string resources for translations.